src/main/java/fr/marcwrobel/jbanking/iban/RandomIban.java

```diff
@@ -0,0 +1,90 @@
+package fr.marcwrobel.jbanking.iban;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Random;
+
+/**
+ * Generates pseudorandom {@link Iban IBANs}.
+ *
+ * <p>
+ * Usage:
+ *
+ * <pre>
+ * // Generating a random IBAN
+ * Iban random1 = new RandomIban().next();
+ *
+ * // Generating a random IBAN using a given Random (in order to make the generation deterministic)
+ * Iban random2 = new RandomIban(new Random(0)).next();
+ *
+ * // Generating a random french or german IBAN
+ * Iban random3 = new RandomIban().next(BbanStructure.FR, BbanStructure.DE);
+ * </pre>
+ *
+ * <p>
+ * This class is not thread-safe and should be used only for tests. It is generating pseudorandom characters using
+ * {@link Random}. Note that Random was choose over SecureRandom because security does not matter in our case and
+ * because Random :
+ * <ul>
+ * <li>produces the same result on all platforms,</li>
+ * <li>produces the same results for a seed by default,</li>
+ * <li>is random enough,</li>
+ * <li>and is much faster.</li>
+ * </ul>
+ */
+public final class RandomIban {
+
+  private final Random random;
+
+  /**
+   * Creates a new random Iban generator using the given {@link Random random number generator}.
+   *
+   * @param random a non-null {@link Random} instance
+   * @throws NullPointerException if the given {@link Random} instance is {@code null}
+   */
+  public RandomIban(Random random) {
+    this.random = requireNonNull(random);
+  }
```

Comment:
Do you use any `@NotNull` annotations in jbanking? They really help with integration with Kotlin.

```java
  public RandomIban(@NotNull Random random) {
    this.random = requireNonNull(random);
  }
```

Reply:
I do not use `@NotNull` annotations in jbanking. I preferred to keep the library completely dependency-free so far.

And I looked at [which `@NotNull` annotation to use](https://stackoverflow.com/questions/4963300/which-notnull-java-annotation-should-i-use) in the past and did not find a good answer. So for other projects I generally use what is available in the classpath, but for jbanking I did not know what to do. Do not hesitate if you have inputs to share ;).

Reply:
For zero dependencies you could use `javax.annotation.Nonnull`, although as the post you linked said, JS305 is abandoned. I wouldn't let that bother you though, it's good enough, especially for a 'plain' library like jbanking.

I think https://jspecify.dev/ is the new standard library that lots of companies/libraries collaborated on, so it should be more widely supported - although it is more recent, so it might be a while until it gains momentum.

Reply:
Thanks for the inputs. I opened a new issue to remind me to have a look at it later : https://github.com/marcwrobel/jbanking/issues/225.